backend/src/main/java/cloudflight/integra/backend/service/DishService.java

```diff
@@ -1,38 +1,143 @@
 package cloudflight.integra.backend.service;
 
+import cloudflight.integra.backend.model.Dish;
+import cloudflight.integra.backend.model.Ingredient;
+import cloudflight.integra.backend.model.Meal;
+import cloudflight.integra.backend.model.Recipe;
 import cloudflight.integra.backend.model.dtos.DishDto;
-import cloudflight.integra.backend.repository.initial.IDishRepository;
+import cloudflight.integra.backend.model.mappers.DishMapper;
+import cloudflight.integra.backend.repository.DishRepository;
+import cloudflight.integra.backend.repository.IngredientRepository;
+import cloudflight.integra.backend.repository.MealRepository;
+import cloudflight.integra.backend.repository.RecipeRepository;
+import jakarta.transaction.Transactional;
 import org.springframework.stereotype.Service;
 
+import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
+import java.util.stream.Collectors;
 
 @Service
 public class DishService {
-    private final IDishRepository repo;
+    private final DishRepository dishRepository;
+    private final DishMapper dishMapper;
+    private final MealRepository mealRepository;
+    private final RecipeRepository recipeRepository;
+    private final IngredientRepository ingredientRepository;
 
-    public DishService(IDishRepository repo) { this.repo = repo; }
+    public DishService(DishRepository dishRepository,
+                       DishMapper dishMapper,
+                       MealRepository mealRepository,
+                       RecipeRepository recipeRepository,
+                       IngredientRepository ingredientRepository) {
+        this.dishRepository = dishRepository;
+        this.dishMapper = dishMapper;
+        this.mealRepository = mealRepository;
+        this.recipeRepository = recipeRepository;
+        this.ingredientRepository = ingredientRepository;
+    }
 
     public DishDto create(DishDto dishDto) {
-        dishDto.setId(UUID.randomUUID());
-        return repo.save(dishDto);
+        Dish dish = dishMapper.toEntity(dishDto);
+        Dish savedDish = dishRepository.save(dish);
+        return dishMapper.toDto(savedDish);
     }
 
-    public List<DishDto> getAll() { return repo.findAll(); }
+    @Transactional
```

Comment:
```suggestion
    @Transactional(readOnly = true)
```
this is just as a learning opportunity, it will automatically run this transaction faster and schedule it in a different way, please also apply this to other places in code where this is the case (I won't comment them all)